mme leaks on CI environments are mitigated

### Problem
* currently we see a lot of failures in integration tests on CI
* one cause is
```
E   urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='192.168.60.142', port=3333): Max retries exceeded with url: /stats/switches (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa7ce0751f0>: Failed to establish a new connection: [Errno 111] Connection refused'))
```
* this usually means that pipelined (on port 3333) is not reachable on magma dev

### Observation
* this seems to fail for a couple of tests after another test had a regular test failure
* after roughly 90s follow up tests can reach pipelined again
* in the syslogs we see a SIGTERM on mme that is followed by LeakSanitizer failures and finally after 90s by a SIGKILL on mme, see e.g.,
```
Sep 14 15:10:20 magma-dev mme[217112]: =================================================================
Sep 14 15:10:20 magma-dev mme[217112]: ==217112==ERROR: LeakSanitizer: detected memory leaks
Sep 14 15:10:20 magma-dev mme[217112]: Direct leak of 160 byte(s) in 2 object(s) allocated from:
Sep 14 15:10:20 magma-dev mme[217112]:     #0 0x7f90a45cc787 in operator new[](unsigned long) ../../../../src/libsanitizer/asan/asan_new_delete.cc:107
Sep 14 15:10:20 magma-dev mme[217112]:     #1 0x7f90a356cb73 in fluid_base::BaseOFConnection::OFReadBuffer::read_notify(unsigned short) fluid/base/BaseOFConnection.cc:84
Sep 14 15:10:20 magma-dev mme[217112]:     #2 0x7f90a356cb73 in fluid_base::BaseOFConnection::OFReadBuffer::read_notify(unsigned short) fluid/base/BaseOFConnection.cc:77
Sep 14 15:10:20 magma-dev mme[217112]:     #3 0x7f90a356cb73 in fluid_base::BaseOFConnection::LibEventBaseOFConnection::read_cb(bufferevent*, void*) fluid/base/BaseOFConnection.cc:353
Sep 14 15:10:20 magma-dev mme[217112]: Direct leak of 48 byte(s) in 1 object(s) allocated from:
Sep 14 15:10:20 magma-dev mme[217112]:     #0 0x7f90a45caa06 in __interceptor_calloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:153
Sep 14 15:10:20 magma-dev mme[217112]:     #1 0x7f90a32a7c0f in zlist_new (/lib/x86_64-linux-gnu/libczmq.so.4+0x24c0f)
Sep 14 15:10:20 magma-dev mme[217112]: SUMMARY: AddressSanitizer: 208 byte(s) leaked in 3 allocation(s).
```

### Hypothesis
* after a failed test the environment is cleaned-up by a restart of services
* pipelined is not restarted properly in time because mme is killed after a 90s systemd timeout

### Actions
* improve logging in case of leaks: use `fast_unwind_on_malloc=0` #13940
* mitigate CI failures by suppressing leaks in external libraries #13940
* this should be further analyzed: are the leaks actually caused by magma code or by external libraries?

### Reproduce Leaks
* build magma vm with services
* (remove leak suppression if already deployed - and restart services)
* `sudo tail -f /var/log/syslog | grep "LeakS"`
* start `s1aptests/test_modify_mme_config_for_sanity.py` (after the test was executed a restart seems to suffice to reproduce a leak - stateful?)
* (add `fast_unwind_on_malloc=0` for a more detailed leak stack trace)
* detailed stack trace with `fast_unwind_on_malloc=0`:
```
  11513 Sep 15 14:54:55 magma-dev mme[45723]: =================================================================
  11514 Sep 15 14:54:55 magma-dev mme[45723]: ==45723==ERROR: LeakSanitizer: detected memory leaks
  11515 Sep 15 14:54:55 magma-dev mme[45723]: Direct leak of 80 byte(s) in 1 object(s) allocated from:
  11516 Sep 15 14:54:55 magma-dev mme[45723]:     #0 0x7fd1dbcb2787 in operator new[](unsigned long) ../../../../src/libsanitizer/asan/asan_new_delete.cc:107
  11517 Sep 15 14:54:55 magma-dev mme[45723]:     #1 0x7fd1dac52b73 in fluid_base::BaseOFConnection::OFReadBuffer::read_notify(unsigned short) fluid/base/BaseOFConnection.cc:84
  11518 Sep 15 14:54:55 magma-dev mme[45723]:     #2 0x7fd1dac52b73 in fluid_base::BaseOFConnection::OFReadBuffer::read_notify(unsigned short) fluid/base/BaseOFConnection.cc:77
  11519 Sep 15 14:54:55 magma-dev mme[45723]:     #3 0x7fd1dac52b73 in fluid_base::BaseOFConnection::LibEventBaseOFConnection::read_cb(bufferevent*, void*) fluid/base/BaseOFConnection.cc:353
  11520 Sep 15 14:54:55 magma-dev mme[45723]:     #4 0x7fd1d90fd827 in bufferevent_run_readcb_ (/lib/x86_64-linux-gnu/libevent-2.1.so.7+0x16827)
  11521 Sep 15 14:54:55 magma-dev mme[45723]:     #5 0x7fd1d910230b  (/lib/x86_64-linux-gnu/libevent-2.1.so.7+0x1b30b)
  11522 Sep 15 14:54:55 magma-dev mme[45723]:     #6 0x7fd1d910813e  (/lib/x86_64-linux-gnu/libevent-2.1.so.7+0x2113e)
  11523 Sep 15 14:54:55 magma-dev mme[45723]:     #7 0x7fd1d910887e in event_base_loop (/lib/x86_64-linux-gnu/libevent-2.1.so.7+0x2187e)
  11524 Sep 15 14:54:55 magma-dev mme[45723]:     #8 0x7fd1dac527e4 in fluid_base::EventLoop::run() fluid/base/EventLoop.cc:64
  11525 Sep 15 14:54:55 magma-dev mme[45723]:     #9 0x7fd1dac527e4 in fluid_base::EventLoop::run() fluid/base/EventLoop.cc:54
  11526 Sep 15 14:54:55 magma-dev mme[45723]:     #10 0x7fd1dac5282c in fluid_base::EventLoop::thread_adapter(void*) fluid/base/EventLoop.cc:76
  11527 Sep 15 14:54:55 magma-dev mme[45723]:     #11 0x7fd1db97c608 in start_thread /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477
  11528 Sep 15 14:54:55 magma-dev mme[45723]:     #12 0x7fd1d96d8132 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x11f132)
  11529 Sep 15 14:54:55 magma-dev mme[45723]: Direct leak of 48 byte(s) in 1 object(s) allocated from:
  11530 Sep 15 14:54:55 magma-dev mme[45723]:     #0 0x7fd1dbcb0a06 in __interceptor_calloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:153
  11531 Sep 15 14:54:55 magma-dev mme[45723]:     #1 0x7fd1da98dc0f in zlist_new (/lib/x86_64-linux-gnu/libczmq.so.4+0x24c0f)
  11532 Sep 15 14:54:55 magma-dev mme[45723]:     #2 0x7fd1da9a9e94 in zsys_init (/lib/x86_64-linux-gnu/libczmq.so.4+0x40e94)
  11533 Sep 15 14:54:55 magma-dev mme[45723]:     #3 0x7fd1da9aa2a6 in zsys_socket (/lib/x86_64-linux-gnu/libczmq.so.4+0x412a6)
  11534 Sep 15 14:54:55 magma-dev mme[45723]:     #4 0x7fd1da99720e in zsock_new_checked (/lib/x86_64-linux-gnu/libczmq.so.4+0x2e20e)
  11535 Sep 15 14:54:55 magma-dev mme[45723]:     #5 0x7fd1da99a491 in zsock_new_pull_checked (/lib/x86_64-linux-gnu/libczmq.so.4+0x31491)
  11536 Sep 15 14:54:55 magma-dev mme[45723]:     #6 0x55a185c9e0e0 in init_task_context /home/vagrant/magma/lte/gateway/c/core/oai/lib/itti/intertask_interface.c:214
  11537 Sep 15 14:54:55 magma-dev mme[45723]:     #7 0x55a181ac18ec in shared_log_thread /home/vagrant/magma/lte/gateway/c/core/oai/common/shared_ts_log.c:185
  11538 Sep 15 14:54:55 magma-dev mme[45723]:     #8 0x7fd1db97c608 in start_thread /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477
  11539 Sep 15 14:54:55 magma-dev mme[45723]:     #9 0x7fd1d96d8132 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x11f132)
  11540 Sep 15 14:54:55 magma-dev mme[45723]: SUMMARY: AddressSanitizer: 128 byte(s) leaked in 2 allocation(s).
```

--- a/lte/gateway/deploy/roles/dev_common/tasks/main.yml
+++ b/lte/gateway/deploy/roles/dev_common/tasks/main.yml
@@ -70,7 +70,17 @@
     -  XDG_CACHE_HOME="{{magma_root}}/.cache"
     -  SWAGGER_CODEGEN_JAR={{ swagger_codegen_jar }}
     -  CODEGEN_ROOT={{ codegen_root }}
-    -  ASAN_OPTIONS=abort_on_error=1:disable_coredump=0:unmap_shadow_on_exit=1
+    -  ASAN_OPTIONS=abort_on_error=1:disable_coredump=0:unmap_shadow_on_exit=1:fast_unwind_on_malloc=0
+    -  LSAN_OPTIONS=suppressions=/usr/local/suppressions.txt
+  when: full_provision
+
+- name: Create LSAN suppressions file
+  copy:
+    dest: "/usr/local/suppressions.txt"
+    content: |
+      leak:libczmq.so.4
+      leak:libzmq.so.5
+      leak:fluid_base::BaseOFConnection::OFReadBuffer::read_notify
   when: full_provision
 
 - name: "Go export"

